opm/output/eclipse/RestartIO.cpp

```diff
@@ -249,8 +255,9 @@ RestartValue load( const std::string& filename,
 
         if (ecl_file_view_has_kw( file_view , key.c_str())) {
             const ecl_kw_type * ecl_kw = ecl_file_view_iget_named_kw( file_view , key.c_str() , 0 );
-            std::vector<double> data = double_vector( ecl_kw );
-            rst_value.extra[ key ] = std::move(data);
+            const double * data_ptr = ecl_kw_get_double_ptr( ecl_kw );
```

Comment:
Ok I think I get your point: extra keywords are read as double, but we no longer use this path to read the now-optional keywords so it is ok. For regular (non-extra) keywords (optional or not) we *do* read float or double interchangeably. Correct?